cpp/mcap/include/mcap/visibility.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#if defined _WIN32 || defined __CYGWIN__
+#  ifdef __GNUC__
+#    define MCAP_EXPORT __attribute__((dllexport))
+#    define MCAP_IMPORT __attribute__((dllimport))
+#  else
+#    define MCAP_EXPORT __declspec(dllexport)
+#    define MCAP_IMPORT __declspec(dllimport)
+#  endif
+// MCAP_BUILDING_DLL to be defined by build process producing a shared library
```

Comment:
Given this comment, it's not clear to me why I seemingly had to define `MCAP_BUILDING_DLL` for the test binaries to build in https://github.com/foxglove/mcap/pull/713 . Part of the problem is how opaque conan is about what is getting built as what, but my understanding is that `MCAP_BUILDING_DLL` is set for both the MCAP library and the consuming executable.

Reply:
in that context i think i'm not even building a DLL, i'm just including the source as a header-only library.

Reply:
Discussed offline and will use `MCAP_IMPLEMENTATION` for `dllexport` since in an exe it doesn't do anything, and otherwise it's needed